src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/ExportProviderBuilder.cs

```diff
@@ -39,34 +34,31 @@ public static async Task<ExportProvider> CreateExportProviderAsync(IEnumerable<s
             Assembly.LoadFrom(path);
         }
 
+        // DevKit assemblies are not shipped in the main language server folder
+        // and not included in ExtensionAssemblyPaths (they get loaded into the default ALC).
+        // So manually add them to the MEF catalog here.
+        if (extensionAssemblyManager.DevKitDependencyPath != null)
+        {
+            assemblyPaths = assemblyPaths.Concat(extensionAssemblyManager.DevKitDependencyPath);
+        }
+
+        // Add the extension assemblies to the MEF catalog.
+        assemblyPaths = assemblyPaths.Concat(extensionAssemblyManager.ExtensionAssemblyPaths);
+
+        ValidateNoDuplicateAssemblies(assemblyPaths, logger);
```

Comment:
Also as a thought: if you moved this logic into the ExtensionAssemblyMangaer.Create() method, then since you're already loading assembly names from paths, you could easily update that to have the assembly name -> path map to get rid of your "loop over all ALCs".